internal/state/store.go

```diff
@@ -144,10 +198,45 @@ func NewStore(db dbm.DB, options StoreOptions) Store {
 	if options.Metrics == nil {
 		options.Metrics = NopMetrics()
 	}
-	return dbStore{
+
+	store := dbStore{
 		db:           db,
 		StoreOptions: options,
 	}
+
+	empty, _ := IsEmpty(store)
+
+	if empty {
+		store.DBKeyLayout = v2Layout{}
+		if err := store.db.SetSync([]byte("version"), []byte("2")); err != nil {
+			panic(err)
+		}
+		if options.Logger != nil {
+			options.Logger.Info("State store version ", "version", "v2")
+		}
+		return store
+	}
+
+	versionNum, err := db.Get([]byte("version"))
+	if len(versionNum) == 0 && err == nil {
```

Comment:
I think we should panic if `err != nil` similar to evidence

Reply:
7e4844ec8 added it but added a warning that in case the DB starts throwing errors on key not found this should not panic.